tempto-core/src/main/java/com/teradata/tempto/internal/convention/tabledefinitions/FileBasedHiveDataSource.java

```diff
@@ -49,26 +49,31 @@ public String getPathSuffix()
     @Override
     public Collection<RepeatableContentProducer> data()
     {
-        Path dataFile = tableDefinitionDescriptor.getDataFile();
-
-        if (!exists(dataFile)) {
-            throw new IllegalStateException("Data file " + tableDefinitionDescriptor.getDataFile() + " should exist");
-        }
+        return tableDefinitionDescriptor.getDataFile()
+                .map(dataFile -> singleton(asRepeatableContentProvider(dataFile)))
```

Comment:
Instead of `singleton` i would use `ImmutableSet.of()`. `singleton` was confusing for me.